js/src/revision-notes.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * Set up a new panel in the post edit sidebar to enable the user to save a revision note as post meta.
+ *
+ * Based on code taken from the block editor handbook.
+ */
+( function( wp ) {
+	let { registerPlugin }         = wp.plugins;
+	let PluginDocumentSettingPanel = wp.editPost.PluginDocumentSettingPanel;
+	let __                         = wp.i18n.__;
+	let TextArea                   = wp.components.TextareaControl;
```

Comment:
I'd recommend against aliasing the component here, it makes the code a little harder to follow.